domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/gtfs/ShapePoint.java

```diff
@@ -0,0 +1,258 @@
+     * field shape_pt_sequence; otherwise return false
+     *
+     * @param otherShapePoint shape to compare
+     * @return true if this {@link ShapePoint} has field shape_pt_sequence greater than {@param otherShape}
+     * field shape_pt_sequence; otherwise return false
+     */
+    public boolean isGreaterThan(final ShapePoint otherShapePoint) {
+        return compareTo(otherShapePoint) > 0;
+    }
+
+    /**
+     * Builder class to create {@link ShapePoint} objects. Allows an unordered definition of the different attributes of
+     * {@link ShapePoint}.
+     */
+    public static class ShapeBuilder {
+        private String shapeId;
+        private Float shapePtLat;
+        private Float shapePtLon;
+        private Integer shapePtSequence;
+        private final List<Notice> noticeCollection = new ArrayList<>();
+        private Float shapeDistTraveled;
+
+        /**
+         * Sets field shapeId value and returns this
+         *
+         * @param shapeId identifies a shape
+         * @return builder for future object creation
+         */
+        public ShapeBuilder shapeId(@NotNull final String shapeId) {
+            this.shapeId = shapeId;
+            return this;
+        }
+
+        /**
+         * Sets field shapePtLat value and returns this
+         *
+         * @param shapePtLat latitude of a shape point
+         * @return builder for future object creation
+         */
+        public ShapeBuilder shapePtLat(@NotNull final Float shapePtLat) {
+            this.shapePtLat = shapePtLat;
+            return this;
+        }
+
+        /**
+         * Sets field shapePtLon value and returns this
+         *
+         * @param shapePtLon longitude of a shape point
+         * @return builder for future object creation
+         */
+        public ShapeBuilder shapePtLon(@NotNull final Float shapePtLon) {
+            this.shapePtLon = shapePtLon;
+            return this;
+        }
+
+        /**
+         * Sets field shapePtSequence value and returns this
+         *
+         * @param shapePtSequence sequence in which the shape points connect to form the shape
+         * @return builder for future object creation
+         */
+        public ShapeBuilder shapePtSequence(@NotNull final Integer shapePtSequence) {
+            this.shapePtSequence = shapePtSequence;
+            return this;
+        }
+
+        /**
+         * Sets field shapeDistTraveled value and returns this
+         *
+         * @param shapeDistTraveled actual distance traveled along the shape from the first shape point to the point
+         *                          specified in this record
+         * @return builder for future object creation
+         */
+        public ShapeBuilder shapeDistTraveled(@Nullable final Float shapeDistTraveled) {
+            this.shapeDistTraveled = shapeDistTraveled;
+            return this;
+        }
+
+        /**
```

Comment:
I left this comment on another entity in a different PR - if the `Builder` here is used as a singleton (single instance), it seems like there should be a `Builder.clear()` method that resets all the fields. Otherwise you're hoping that the calling class will know to reset all of the fields manually themselves, or always call all of the builder fields and pass in `null` values when the field is not provided.

Reply:
@fabrice-v, opening a issue on this matter since all gtfs entities will be affected by such change.

Reply:
See https://github.com/MobilityData/gtfs-validator/issues/290